.github/workflows/_rust.yml

```diff
@@ -143,22 +149,35 @@ jobs:
         run: pnpm install
       - name: Build release exe and MSI
         run: pnpm build
-      - name: Compute SHA256 and rename
-        run: |
-          # Used for release artifact
-          # In release mode the name comes from tauri.conf.json
-          cp "../target/release/Firezone.exe" "${{ env.BINARY_DEST_PATH }}-x64.exe"
-          cp "../target/release/bundle/msi/*.msi" "${{ env.BINARY_DEST_PATH }}-x64.msi"
-          cp "../target/release/firezone_gui_client.pdb" "${{ env.BINARY_DEST_PATH }}-x64.pdb"
-
-          Get-FileHash ${{ env.BINARY_DEST_PATH }}-x64.exe -Algorithm SHA256 | Select-Object Hash > ${{ env.BINARY_DEST_PATH }}-x64.exe.sha256sum.txt
-          Get-FileHash ${{ env.BINARY_DEST_PATH }}-x64.msi -Algorithm SHA256 | Select-Object Hash > ${{ env.BINARY_DEST_PATH }}-x64.msi.sha256sum.txt
-          Get-FileHash ${{ env.BINARY_DEST_PATH }}-x64.pdb -Algorithm SHA256 | Select-Object Hash > ${{ env.BINARY_DEST_PATH }}-x64.pdb.sha256sum.txt
-
-          # This might catch regressions in #3384, depending how CI runners
-          # handle exit codes
-          git diff --exit-code
-      - name: Save GUI client
+      - name: Rename artifacts and compute SHA256
+        shell: bash
+        run: ${{ matrix.rename-script }}
+      - name: Save Linux client
+        if: ${{ runner.os == 'Linux' }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ env.BINARY_DEST_PATH }}-amd64
+          path: |
+            ${{ github.workspace }}/rust/gui-client/${{ env.BINARY_DEST_PATH }}-amd64.exe
+            ${{ github.workspace }}/rust/gui-client/${{ env.BINARY_DEST_PATH }}-amd64.exe.sha256sum.txt
+      - name: Save Linux AppImage
+        if: ${{ runner.os == 'Linux' }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ env.BINARY_DEST_PATH }}_amd64.AppImage
+          path: |
+            ${{ github.workspace }}/rust/gui-client/${{ env.BINARY_DEST_PATH }}_amd64.AppImage
+            ${{ github.workspace }}/rust/gui-client/${{ env.BINARY_DEST_PATH }}_amd64.AppImage.sha256sum.txt
+      - name: Save Linux deb package
+        if: ${{ runner.os == 'Linux' }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ env.BINARY_DEST_PATH }}_amd64.deb
+          path: |
+            ${{ github.workspace }}/rust/gui-client/${{ env.BINARY_DEST_PATH }}_amd64.deb
+            ${{ github.workspace }}/rust/gui-client/${{ env.BINARY_DEST_PATH }}_amd64.deb.sha256sum.txt
+      - name: Save Windows client
```

Comment:
I couldn't find a simple way to upload 3 artifacts over 2 OSes without all this duplication. DDG search didn't return anything obvious and ChatGPT said it's impossible. Someone on SO suggested using a job matrix but then I'd be building the program 3 times, or tarring up the directory and handing it off between runners or something.